ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiParticipantPruningService.scala

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.services.admin
+
+import java.util.UUID
+
+import com.daml.dec.DirectExecutionContext
+import com.daml.ledger.api.v1.admin.participant_pruning_service.{
+  ParticipantPruningServiceGrpc,
+  PruneRequest
+}
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.ledger.participant.state.index.v2.{IndexParticipantPruningService, LedgerEndService}
+import com.daml.ledger.participant.state.v1.{
+  NotPruned,
+  Offset,
+  ParticipantPruned,
+  SubmissionId,
+  WriteParticipantPruningService
+}
+import com.daml.lf.data.Ref
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.platform.api.grpc.GrpcApiService
+import com.daml.platform.server.api.validation.ErrorFactories
+import com.google.protobuf.empty.Empty
+import io.grpc.{ServerServiceDefinition, StatusRuntimeException}
+
+import scala.compat.java8.FutureConverters
+import scala.concurrent.{ExecutionContext, Future}
+
+final class ApiParticipantPruningService private (
+    readBackend: IndexParticipantPruningService with LedgerEndService,
+    writeBackend: WriteParticipantPruningService)(
+    implicit grpcExecutionContext: ExecutionContext,
+    logCtx: LoggingContext)
+    extends ParticipantPruningServiceGrpc.ParticipantPruningService
+    with GrpcApiService {
+
+  private val logger = ContextualizedLogger.get(this.getClass)
+
+  override def bindService(): ServerServiceDefinition =
+    ParticipantPruningServiceGrpc.bindService(this, DirectExecutionContext)
```

Comment:
Please coordinate with @SamirTalwar-DA on [this PR](https://github.com/digital-asset/daml/pull/7945).

Reply:
Yeah, I have been monitoring that PR.

Reply:
As this one is likely to make it in first, I suggest you just comment on the other PR once it's merged so I don't forget to update this. (Or push an update there, if you like.)